lib/matplotlib/backend_tools.py

```diff
@@ -0,0 +1,948 @@
+"""
+Abstract base classes define the primitives for Tools.
+These tools are used by `NavigationBase`
+
+:class:`ToolBase`
+    Simple stateless tool
+
+:class:`ToolToggleBase`
+    Tool that has two states, only one Toggle tool can be
+    active at any given time for the same `Navigation`
+"""
+
+
+from matplotlib import rcParams
+from matplotlib._pylab_helpers import Gcf
+import matplotlib.cbook as cbook
+from weakref import WeakKeyDictionary
+import numpy as np
+
+
+class Cursors(object):
+    """Simple namespace for cursor reference"""
+    HAND, POINTER, SELECT_REGION, MOVE = list(range(4))
+cursors = Cursors()
+
+# Views positions tool
+_views_positions = 'viewpos'
+
+
+class ToolBase(object):
```

Comment:
Is it worth doing this as an abstract base class?  That way you can force the implementer to over-ride a set of functions.